core/src/main/scala/kafka/admin/LeaderElectionCommand.scala

```diff
@@ -0,0 +1,235 @@
+            if (cause.isInstanceOf[TimeoutException]) {
+              println("Timeout waiting for election results")
+            }
+            throw new AdminCommandFailedException("Timeout waiting for election results", cause)
+          case e: Throwable =>
+            println("Error while making request")
+            e.printStackTrace()
+            return
+      }
+    }
+
+    val (failed, succeeded) = attemptedPartitions
+      .map(topicPartition => topicPartition -> result.partitionResult(topicPartition))
+      .partition { case (_, partitionResult) => completedExceptionally(partitionResult) }
+
+    if (!succeeded.isEmpty) {
+      val partitions = succeeded.map(_._1).mkString(", ")
+      println(s"Successfully completed leader election ($electionType) for partitions $partitions")
+    }
+
+    if (!failed.isEmpty) {
+      val rootException = new AdminCommandFailedException(s"${failed.size} replica(s) could not be elected")
+      failed.foreach { case (topicPartition, future) =>
+        val exception = Try(future.get()).fold(
+          identity,
+          _ => new AdminCommandFailedException("Exceptional future with no exception")
+        )
+        println(s"Error completing leader election ($electionType) for partition: $topicPartition: $exception")
+        rootException.addSuppressed(exception)
+      }
+      throw rootException
+    }
+  }
+
+  private[this] def completedExceptionally(future: KafkaFuture[_]): Boolean = {
+    try {
+      future.get()
+      false
+    } catch {
+      case (_: Throwable) =>
+        true
+    }
+  }
+}
+
+private final class LeaderElectionCommandOptions(args: Array[String]) extends CommandDefaultOptions(args) {
+  val bootstrapServer = parser
+    .accepts(
+      "bootstrap-server",
+      "A hostname and port for the broker to connect to, in the form host:port. Multiple comma separated URLs can be given. REQUIRED.")
+    .withRequiredArg
+    .required
+    .describedAs("host:port")
+    .ofType(classOf[String])
+  val adminClientConfig = parser
+    .accepts(
+      "admin.config",
+      "Configuration properties files to pass to the admin client")
+    .withRequiredArg
+    .describedAs("config file")
+    .ofType(classOf[String])
+
+  val pathToJsonFile = parser
+    .accepts(
+      "path-to-json-file",
+      "The JSON file with the list  of partition for which leader elections should be performed. This is an example format. \n{\"partitions\":\n\t[{\"topic\": \"foo\", \"partition\": 1},\n\t {\"topic\": \"foobar\", \"partition\": 2}]\n}\nNot allowed if --all-topic-partitions or --topic flags are specified.")
+    .withRequiredArg
+    .describedAs("Path to JSON file")
+    .ofType(classOf[String])
+
+  val topic = parser
+    .accepts(
+      "topic",
+      "Name of topic for which to perform an election. Not allowed if --path-to-json-file or --all-topic-partitions is specified.")
+    .availableUnless("path-to-json-file")
+    .withRequiredArg
```

Comment:
Should this be `requiredIf("partition")`?

Reply:
That tried that at first. I haven't thought about the logic to implement something like this but unfortunately jopts-simple implementation of this required logic is limited. TL;DR; using `requiredIf` didn't work because I think there is a cycle in the graph.